docker/mongodb-agent/dummy-probe.sh

```diff
@@ -1,5 +1,37 @@
 #!/bin/bash
-# Dummy liveness probe that returns success to keep container alive during script copying
-# This prevents container from being killed while waiting for real probe scripts
-echo "Using dummy liveness probe - keeping container alive until real probe script is copied"
-exit 0
+# Dynamic liveness probe that locates and executes the actual probe from init-database container
+
+find_init_container() {
+  local pid
+  pid=$(pgrep -f "agent-utilities-holder_marker" | head -n1)
+  if [[ -n "$pid" && -d "/proc/$pid/root/probes" ]]; then
+    echo "$pid"
+    return 0
+  fi
+  return 1
+}
+
+execute_liveness_probe() {
+  local init_pid="$1"
+  local init_probe_path="/proc/$init_pid/root/probes/probe.sh"
+
+  if [[ -f "$init_probe_path" && -x "$init_probe_path" ]]; then
+    # Execute the actual probe script from the init-database container
+    # This works because of shared process namespace - the probe can see all processes
+    exec "$init_probe_path"
+  else
+    echo "ERROR: Liveness probe script not found or not executable at $init_probe_path"
```

Comment:
can we split this message into two distinct error messages - one for `Liveness probe script not found` and one for `not executable`? This might help in future with debugging of what is happening